### DISCUSSION TITLE
Loading UI based on a cookie value partially renders the change

### DISCUSSION THREAD

**Discussion by daecepi** [vercel/next.js#37696] [Help]
Has anyone come accross the scenario of:
Loading some personalized components based on a cookie value but when it is compiling no cookies will be available (in server side) so the default UI renders but then in the client when the cookie is identified only text changes and things like:
- <Link does not update its href= property
- <div does not apply specific className

So the full UI ends up partially updated and with bugs

Even when you are depending on useStates and useEffects to change the UI (so it should?)

**Comment by daecepi** [comment]
- className specifics do not update appropiately
- Link component does not update its href property

**Comment by vadymshymko** [comment]
Hello @daecepi. I recommend trying one of the following 2 approaches:
1. Read cookies inside getServerSideProps (from context.req.headers.cookie ), and then push them to the client side via props
```javascript
// some function for reading cookies from a string
function getCookieValueFromString(name, str = '') {
  //...some code
  return someValue;
}

async function getServerSideProps(context) {
  const someCookie = getCookieValueFromString(context.req.headers.cookie);
  
  return {
    props: {
      someProp: someCookie
    }
  }
}


import Link from 'next/link';

function SomeComponent({someProp}) {
  
  return (
     <div>
       <div className={someProp === 'foo' ? 'bar' : 'baz'} />
       
       <Link href={someProp === 'foo' ? '/path1' : 'path2'}}>
         <a>Link text</a>
       <Link>
     </div>
  )
}
```

2. Use/recheck your implementation of useState and useEffect

```javascript
import Link from 'next/link';

function SomeComponent() {
  const [someCookieValue, setSomeCookieValue] = useState();
  
  useEffect(() =>{
    const cookieValue = // some code for get cookie value here..
    setSomeCookieValue(cookieValue);
  }, []);
  
  return (
     <div>
       <div className={someCookieValue === 'foo' ? 'bar' : 'baz'} />
       
       <Link href={someCookieValue === 'foo' ? '/path1' : 'path2'}}>
         <a>Link text</a>
       <Link>
     </div>
  )
}
```

Hope this helps)
